src/OSPSuite.Presentation/Presenters/Charts/ChartEditorPresenter.cs

```diff
@@ -474,23 +482,52 @@ public void SetCurveNameDefinition(Func<DataColumn, string> curveNameDefinition)
          _curveSettingsPresenter.CurveNameDefinition = curveNameDefinition;
       }
 
+      public void AddCurvesWithSameColorForColumn(IReadOnlyList<DataColumn> dataColumnList, CurveOptions defaultCurveOptions = null)
+      {
+         var groupColor = Chart.SelectNewColor();
+         foreach (var dataColumn in dataColumnList)
+         {
+            var (exists, curve) = createAndConfigureCurve(dataColumn, defaultCurveOptions);
+
+            if (exists) continue;
+
+            if (defaultCurveOptions != null)
+               curve.CurveOptions.UpdateFrom(defaultCurveOptions);
+
+            curve.Color = groupColor;
+            curve.UpdateStyleForObservedData();
+
+            Chart.AddCurve(curve);
+         }
+      }
+
+
       public Curve AddCurveForColumn(DataColumn dataColumn, CurveOptions defaultCurveOptions = null)
       {
-         var curve = Chart.CreateCurve(dataColumn.BaseGrid, dataColumn, _curveNameDefinition(dataColumn), _dimensionFactory);
+         var (exists, curve) = createAndConfigureCurve(dataColumn, defaultCurveOptions);
 
-         if (Chart.HasCurve(curve.Id))
-            return Chart.CurveBy(curve.Id);
+         if (exists) return curve;
```

Comment:
not on one line

Reply:
@georgeDaskalakis you can change this in the next PR

Reply:
done